src/eslint_result.ts

```diff
@@ -0,0 +1,98 @@
+import * as core from "@actions/core"
+import { ChangeRange, generateChangeRanges } from "./git_utils"
+
+export type ResultObject = {
+  filePath: string
+  messages: EslintMessage[]
+}
+
+export type EslintResultInstance = InstanceType<typeof EslintResult>
+
+type EslintMessage = {
+  ruleId: string
+  severity: number
+  message: string
+  line: number
+  column: number
+  nodeType: string
+  messageId: string
+  endLine: number
+  endColumn: number
+  suggestions: any[]
+}
+
+export class EslintResult {
+  public relevantWarningCount: number = 0
+  public relevantErrorCount: number = 0
+  private resultObject: ResultObject
+  private compareSha: string
+  private changeRanges?: ChangeRange[]
+  private relevantMessages: EslintMessage[] = []
+
+  constructor(resultObject: ResultObject, compareSha: string) {
+    this.resultObject = resultObject
+    this.compareSha = compareSha
+  }
+
+  async asyncInitialize() {
+    this.changeRanges = await generateChangeRanges(
+      this.resultObject.filePath,
+      this.compareSha,
+    )
+    this.findRelevantMessages()
+    this.calculateCounts()
+  }
+
+  outputAnnotations() {
+    this.relevantMessages.forEach((msg) => {
+      let options: core.AnnotationProperties = {
+        title: msg.ruleId,
+        file: this.repoFilePath,
+        startLine: msg.line,
+        endLine: msg.endLine,
+        startColumn: msg.column,
+        endColumn: msg.endColumn,
+      }
+      switch (msg.severity) {
+        case 1:
```

Comment:
Purely for readability, maybe let's extract an enum for severity

```ts
enum Severity {
    Warning = 1,
    Error = 2
}
```

Reply:
Great call 👍